projectile.el

```diff
@@ -6272,14 +6272,19 @@ Otherwise behave as if called interactively.
       (projectile-discover-projects-in-search-path))
     (add-hook 'project-find-functions #'project-projectile)
     (add-hook 'find-file-hook 'projectile-find-file-hook-function)
-    (add-hook 'projectile-find-dir-hook #'projectile-track-known-projects-find-file-hook t)
-    (add-hook 'dired-before-readin-hook #'projectile-track-known-projects-find-file-hook t t)
+    (if (version<= "28.1" emacs-version)
```

Comment:
I'd add a comment here explaining the need for this check and the hook in general.  (e.g. something similar to what you've written in the PR's description)

Reply:
Great point, I've gone ahead and added a summary. I can expand on it a bit more and include a link to the (relevant NEWS entry)[https://git.savannah.gnu.org/cgit/emacs.git/tree/etc/NEWS.28?h=master#n2956] but hopefully this is clear enough.